google/cloud/internal/rest_stub_helpers.h

```diff
@@ -58,7 +60,9 @@ StatusOr<Response> RestResponseToProto(RestResponse&& rest_response) {
   return destination;
 }
 
-template <typename Request>
+template <
+    typename Response, typename Request,
+    std::enable_if_t<std::is_same<Response, EmptyResponseType>::value, int> = 0>
 Status Delete(
     rest_internal::RestClient& client, rest_internal::RestContext& rest_context,
     Request const&, bool, std::string path,
```

Comment:
comment that predates this PR: the unused request parameter looked strange to me, but I see that
- the contents of `request` are used to make `path`
- it lets us consolidate generator code